src/VisualStudio/IntegrationTest/TestUtilities/Input/AbstractSendKeys.cs

```diff
@@ -0,0 +1,202 @@
+            if (result == -1)
+            {
+                // This is a unicode character that must be handled differently.
+
+                AddUnicodeInputs(inputs, ch);
+                return;
+            }
+
+            var virtualKey = (VirtualKey)(result & 0xff);
+            var shiftState = (ShiftState)(((ushort)result >> 8) & 0xff);
+
+            AddInputs(inputs, virtualKey, shiftState);
+        }
+
+        private static void AddUnicodeInputs(List<NativeMethods.INPUT> inputs, char ch)
+        {
+            var keyDownInput = new NativeMethods.INPUT
+            {
+                Type = NativeMethods.INPUT_KEYBOARD,
+                ki = new NativeMethods.KEYBDINPUT
+                {
+                    wVk = 0,
+                    wScan = ch,
+                    dwFlags = NativeMethods.KEYEVENTF_UNICODE,
+                    time = 0,
+                    dwExtraInfo = IntPtr.Zero
+                }
+            };
+
+            var keyUpInput = new NativeMethods.INPUT
+            {
+                Type = NativeMethods.INPUT_KEYBOARD,
+                ki = new NativeMethods.KEYBDINPUT
+                {
+                    wVk = 0,
+                    wScan = ch,
+                    dwFlags = NativeMethods.KEYEVENTF_UNICODE | NativeMethods.KEYEVENTF_KEYUP,
+                    time = 0,
+                    dwExtraInfo = IntPtr.Zero
+                }
+            };
+
+            inputs.Add(keyDownInput);
+            inputs.Add(keyUpInput);
+        }
+
+        private static void AddInputs(List<NativeMethods.INPUT> inputs, VirtualKey virtualKey, uint dwFlags)
+        {
+            NativeMethods.INPUT input;
+            var scanCode = NativeMethods.MapVirtualKey((uint)virtualKey, NativeMethods.MAPVK_VK_TO_VSC);
+            if (scanCode != 0)
+            {
+                input = new NativeMethods.INPUT
+                {
+                    Type = NativeMethods.INPUT_KEYBOARD,
+                    ki = new NativeMethods.KEYBDINPUT
+                    {
+                        wVk = 0,
+                        wScan = (ushort)scanCode,
+                        dwFlags = dwFlags | NativeMethods.KEYEVENTF_SCANCODE,
+                        time = 0,
+                        dwExtraInfo = IntPtr.Zero
+                    }
+                };
+            }
+            else
+            {
+                input = new NativeMethods.INPUT
+                {
+                    Type = NativeMethods.INPUT_KEYBOARD,
+                    ki = new NativeMethods.KEYBDINPUT
+                    {
+                        wVk = (ushort)virtualKey,
+                        wScan = 0,
+                        dwFlags = dwFlags,
+                        time = 0,
+                        dwExtraInfo = IntPtr.Zero
+                    }
+                };
+            }
+
+            if (IsExtendedKey(virtualKey))
+            {
+                input.ki.dwFlags |= NativeMethods.KEYEVENTF_EXTENDEDKEY;
```

Comment:
>KEYEVENTF_EXTENDEDKEY [](start = 50, length = 21)

Is this flag valid when using a virtual key? The docs are ambiguous here but the language suggests it's only valid when applied to a scan code.

[https://docs.microsoft.com/en-us/windows/desktop/api/winuser/ns-winuser-tagkeybdinput](https://docs.microsoft.com/en-us/windows/desktop/api/winuser/ns-winuser-tagkeybdinput)

Reply:
It's required for scan codes, making the change itself a bug fix for that case. The use of this flag for other cases matches the behavior that was always used in this library so it's not really a change.